### DISCUSSION TITLE
How to programmatically create a new Delta with complex types

### DISCUSSION THREAD

**Discussion by andrei-ionescu** [delta-io/delta-rs#507] [Q&A]
I'm want to create a new Delta table using `delta-rs` and I'm trying to create the needed schema based on the source dataset but I cannot create the following data types:
- `SchemaTypeMap`
- `SchemaTypeArray`

They don't have a `new` method implementation and their properties are not public.

What's the proper way to programatically create a Delta table using `delta-rs` with complex types like maps and arrays? Or, what's the proper way to create those two complex types?

**Comment by houqp** [comment]
let's focus the discussion in https://github.com/delta-io/delta-rs/issues/513 :)
